plugin/testdata/all-typedefs.lua

```diff
@@ -7,35 +7,37 @@ return {
     { config = {
         type = "record",
         fields = {
-          { http_method = typedefs.http_method },
-          { protocol = typedefs.protocol },
+          { auto_timestamp_ms = typedefs.auto_timestamp_ms },
+          { auto_timestamp_s = typedefs.auto_timestamp_s },
+          { certificate = typedefs.certificate },
+          { cidr = typedefs.cidr },
+          { cidr_v4 = typedefs.cidr_v4 },
+          { destinations = typedefs.destinations },
+          { header_name = typedefs.header_name },
           { host = typedefs.host },
           { host_with_optional_port = typedefs.host_with_optional_port },
-          { wildcard_host = typedefs.wildcard_host },
+          { hosts = typedefs.hosts },
+          { http_method = typedefs.http_method },
           { ip = typedefs.ip },
           { ip_or_cidr = typedefs.ip_or_cidr },
-          { cidr_v4 = typedefs.cidr_v4 },
-          { port = typedefs.port },
-          { path = typedefs.path },
-          { url = typedefs.url },
-          { header_name = typedefs.header_name },
-          { timeout = typedefs.timeout },
-          { uuid = typedefs.uuid },
-          { auto_timestamp_s = typedefs.auto_timestamp_s },
-          { auto_timestamp_ms = typedefs.auto_timestamp_ms },
+          { key = typedefs.key },
+          { lua_code = typedefs.lua_code },
           { name = typedefs.name },
-          { utf8_name = typedefs.utf8_name },
+          { path = typedefs.path },
+          { port = typedefs.port },
+          { protocol = typedefs.protocol },
+          { router_path = typedefs.router_path },
+          { router_paths = typedefs.router_paths },
+          { semantic_version = typedefs.semantic_version },
           { sni = typedefs.sni },
-          { certificate = typedefs.certificate },
-          { key = typedefs.key },
+          { sources = typedefs.sources },
           { tag = typedefs.tag({ required = false }) },
           { tags = typedefs.tags },
-          { sources = typedefs.sources },
-          { destinations = typedefs.destinations },
-          { hosts = typedefs.hosts },
-          { paths = typedefs.paths },
```

Comment:
See the commit message or PR description for what was altered.

However, part of those changes was removing the `typedefs.paths` typedef, as it was removed from Kong. In the event users were to use this typedef on plugin schemas, it would cause issues (however they could easily replicate it with the existing `typedefs.path` typedef.

Are we okay with this removal? We can easily add it back, but should we? This PR is held from merging for this reason. @mikefero stated that he'll look into this a bit further, but figured to get this PR up anyway.

Reply:
We  can't completely get rid off typedefs.path. It is most certainly being used by some plugins.

If we are certain that none of the relevant plugins use this typedef, we can remove it. This includes the custom plugins that matter.

Reply:
`typedefs.paths` is not used anywhere in Koko nor is used in any custom plugins. As such, we could technically remove it, however, that still may not be the right thing to do.

Reply:
from the documentation:
> The typedefs variable (obtained by requiring kong.db.schema.typedefs) is a table containing a lot of useful type definitions and aliases....
> To learn more about schemas, see:
>     ·  The source code of [typedefs.lua](https://github.com/Kong/kong/blob/2.8.0/kong/db/schema/typedefs.lua) to get an idea of what’s provided there by default.

To me this sounds like most typedefs should stay.

Reply:
I am on the fence with this. `path` and `paths` seemed to be historically for routes and the router and most plugins (and custom plugins I have seen) use string for path fields. If we want to keep backwards compatibility for the small (if not 0) percentage of users using this typedef I think we should alias `path` and `paths` to `router_path` and `router_paths`.

Also we need to add additional tests for `router_path` and router_paths` since you are including them here. I noticed the validate functions are calling `ngx.re.find` which is not available in `goks` and the aliasing piece for the loader is still being worked on by @omegabytes.

I think `cidr` is already being handled by `ip_or_cidr` in the tests, but please verify and create a test for that as well.

Reply:
> @mikefero I am on the fence with this. path and paths seemed to be historically for routes and the router and most plugins (and custom plugins I have seen) use string for path fields. If we want to keep backwards compatibility for the small (if not 0) percentage of users using this typedef I think we should alias path and paths to router_path and router_paths.

I'm also on the fence, as the use of this legacy typedef would likely be quite low; I'm not saying removing it is the right answer either. 

I frankly didn't dig in too much to see if aliasing `path` & `paths` to `router_path` & `router_paths` would be 100% backwards compatible, but indeed, that's an option & something we can audit.

I'll take a look into the `cidr`/`ip_or_cidr` thing you mentioned, but for some background, I just wanted to ensure all typedefs were part of this test plugin fixture.

Reply:
Ended up re-introducing `typedefs.path` per the conversations here.